vowpalwabbit/cli_help_formatter.cc

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "cli_help_formatter.h"
+#include <sstream>
+#include <string>
+#include "options.h"
+#include "text_utils.h"
+
+using namespace VW::config;
+
+// Returns the one of values as a comma separated string or empty if there are none.
+std::string one_of_string(base_option& opt)
+{
+  std::string ret_val = "";
+  struct extractor : typed_option_visitor
+  {
+    std::string& m_ret_val;
+    extractor(std::string& ret_val) : m_ret_val(ret_val) {}
+
+    void visit(typed_option<uint32_t>& option) override
+    {
+      if (!option.one_of().empty()) { m_ret_val = fmt::format("{}", fmt::join(option.one_of(), ", ")); }
+    };
+    void visit(typed_option<uint64_t>& option) override
+    {
+      if (!option.one_of().empty()) { m_ret_val = fmt::format("{}", fmt::join(option.one_of(), ", ")); }
+    };
+    void visit(typed_option<int64_t>& option) override
+    {
+      if (!option.one_of().empty()) { m_ret_val = fmt::format("{}", fmt::join(option.one_of(), ", ")); }
+    };
+    void visit(typed_option<int32_t>& option) override
+    {
+      if (!option.one_of().empty()) { m_ret_val = fmt::format("{}", fmt::join(option.one_of(), ", ")); }
+    };
+    void visit(typed_option<bool>& /*option*/) override{};
+    void visit(typed_option<float>& /*option*/) override{};
```

Comment:
is this meant to be different than int, uint, etc..

Reply:
Not sure I understand?

Reply:
I meant this `if (!option.one_of().empty()) { m_ret_val = fmt::format("{}", fmt::join(option.one_of(), ", "));`

But realized it wouldnt make much sense to require specific float values